UofS-Pulse-Binfo/kp_frontpage
# Experiments section

![Screen Shot 2020-06-09 at 11 32 32 AM (2)](https://user-images.githubusercontent.com/15472253/84180593-031e3c80-aa45-11ea-9c38-47ee04b4eae5.png)

This PR will add a Experiment section in Frontpage. The update includes a Tripal Field (highlighted/promote) to allow admin to select a set of projects to be highlighted in this section through the content type control panel.

**Metadata**

Issue # (https://github.com/UofS-Pulse-Binfo/kp_frontpage/issues/15) Add project summary


Documentation:

Testing?

To test:
1. Update the module (kp_frontpage) by uninstalling and re-enabling in the modules list.
This will ensure the system variables referenced in queries are correctly initialized.

2. Add the highligted field to project/experiment/study content type. Use the link "Check for new fields".

3. By default the section will not render in Frontpage when there is no project selected.

4. Create or update a project and set the field to either of the option below:
![Screen Shot 2020-06-09 at 11 51 43 AM (2)](https://user-images.githubusercontent.com/15472253/84182458-aa9c6e80-aa47-11ea-904c-3221d736db13.png)

A. --- - Does not wish to promote experiment to Frontpage.
B. Promote to Frontpage - Highlight project in Experiment section.
C. Promote to Frontpage and set a default - Same as 2 but will rank a project higher than just option # 2 and will put this item first in the list.

Project/Experiment description may contain official logo/logotype. For the module to parse the logo, indicate the image with the class = **kp-frontpage-logo**. For example:
`<img class="kp-frontpage-logo" src="site/project_logo.jpg" alt="Project Logo" width="300" height="138" />
`

Repeat for at least 3 projects per crop or genus.

Note:
1. On small screen or when window is resized only first project in the list becomes available.
Option to select other experiment is disabled as it requires a significant css, html and js manipulations.
2. Twitter page opens in new window.
3. The experiment slider's optimal count to maintain layout is 3 items per slide.
4. A genus is seleted per page load.
